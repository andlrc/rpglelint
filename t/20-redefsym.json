[
   {
      "column":7,
      "file":"t/20-redefsym.rpgle",
      "line":4,
      "message":"redefinition of 'a' as a different kind of symbol",
      "severity":"warning"
   },
   {
      "column":7,
      "file":"t/20-redefsym.rpgle",
      "line":3,
      "message":"previous definition is here",
      "severity":"note"
   },
   {
      "column":9,
      "file":"t/20-redefsym.rpgle",
      "line":9,
      "message":"redefinition of 'c' as a different kind of symbol",
      "severity":"warning"
   },
   {
      "column":9,
      "file":"t/20-redefsym.rpgle",
      "line":8,
      "message":"previous definition is here",
      "severity":"note"
   }
]
