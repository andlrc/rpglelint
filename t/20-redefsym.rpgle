// rpglelint: -Wredefining-symbol

dcl-s a int(10);
dcl-s a varchar(32);
dcl-s b int(10);
dcl-proc proc1;
  dcl-s b varchar(32);
  dcl-s c int(10);
  dcl-s c varchar(32);
end-proc;
