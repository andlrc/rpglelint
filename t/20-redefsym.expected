t/20-redefsym.rpgle:4:7: warning: redefinition of 'a' as a different kind of symbol [-Wredefining-symbol]
 dcl-s a varchar(32);
       ^
t/20-redefsym.rpgle:3:7: note: previous definition is here
 dcl-s a int(10);
       ^
t/20-redefsym.rpgle:9:9: warning: redefinition of 'c' as a different kind of symbol [-Wredefining-symbol]
   dcl-s c varchar(32);
         ^
t/20-redefsym.rpgle:8:9: note: previous definition is here
   dcl-s c int(10);
         ^
